packages/cubejs-teradata-jdbc-driver/src/TeradataQuery.ts

```diff
@@ -0,0 +1,139 @@
+import { BaseFilter, BaseQuery } from '@cubejs-backend/schema-compiler';
+import R, { concat } from 'ramda';
+
+const GRANULARITY_TO_INTERVAL: Record<string, string> = {
+  day: 'DD',
+  hour: 'HH',
+  minute: 'MI',
+  second: 'second',
+  month: 'MONTH',
+  year: 'YEAR'
+};
+
+class TeradataFilter extends BaseFilter {
+  public likeIgnoreCase(column: any, not: any, param: any, type: string) {
+    const p = (!type || type === 'contains' || type === 'ends') ? '%' : '';
+    const s = (!type || type === 'contains' || type === 'starts') ? '%' : '';
+    return `${column}${not ? ' NOT' : ''} LIKE CONCAT('${p}', ${this.allocateParam(param)}, '${s}')`;
+  }
+}
+
+export class TeradataQuery extends BaseQuery {
+  public newFilter(filter: any) {
+    return new TeradataFilter(this, filter);
+  }
+
+  public concatStringsSql(strings: string[]) {
+    return `CONCAT(${strings.join(', ')})`;
+  }
+
+  public convertTz(field: string) {
+    return `${field}`;
+  }
+
+  //  TO CHECK
+  public timeStampCast(value: string) {
+    console.log('timeStamp timeStampCast', value);
+    return `CAST(OREPLACE(OREPLACE(${value}, 'T', ' '), 'Z', '') AS TIMESTAMP(3))`;
+  }
+
+  //  TODO
+  public dateTimeCast(value: string) {
+    console.log('timeStamp dateTimeCast', value);
+    return `CAST(OREPLACE(OREPLACE(${value}, 'T', ' '), 'Z', '') AS TIMESTAMP(3))`;
+  }
+
+  public subtractInterval(date: string, interval: string) {
+    const [number, type] = this.parseInterval(interval);
+    console.log(`(${date} - INTERVAL '${number}' ${type})`);
+    return `(${date} - INTERVAL '${number}' ${type})`;
+  }
+
+  public addInterval(date: string, interval: string) {
+    const [number, type] = this.parseInterval(interval);
+    console.log(`(${date} + INTERVAL '${number}' ${type})`);
+    return `(${date} + INTERVAL '${number}' ${type})`;
+  }
+
+  public timeGroupedColumn(granularity: string, dimension: string): string {
+    console.log(`TRUNC(CAST(${dimension} AS TIMESTAMP(6)), '${GRANULARITY_TO_INTERVAL[granularity]}')`);
+    return `TRUNC(CAST(${dimension} AS TIMESTAMP(6)), '${GRANULARITY_TO_INTERVAL[granularity]}')`;
+  }
+
+  public rowNumberColumn() {
+    return ', ROW_NUMBER() OVER (ORDER BY 1) AS RowNum_ ';
+  }
+
+  public groupByDimensionLimit() {
+    let startWindow = 0;
+    let endWindow = 1000;
+    if (this.offset && parseInt(this.offset, 10)) {
```

Comment:
what's the 10? window size? can we put a const variable for it for clarity?

Reply:
it's the RADIX, I'll put it in a const VARIABLE.